### PR TITLE
deprecate `spawn(cmd)` to `run(cmd, wait=false)`

### LINKED ISSUE #25965: Rename `spawn` function
The `spawn` function is the same as the `run` function but non-blocking, it runs the process in the background.

The better known `@spawn` macro, is used to run a function on a different julia process.

While conceptually related they are not related in any practical sense.

I suggest that `spawn(cmd)` should be deprecated to something like `run_background(cmd)`,
or to `run(cmd; background=true)`

### PR DESCRIPTION
**Pull request by JeffBezanson**
Also add `read` and `write` keyword arguments to `open` for processes.

fixes #25965

### CODE REVIEW

**Review by StefanKarpinski — APPROVED** [review]
I thought I was going to be annoyed by `run` returning the process object when called with `wait=true` but I'm actually not – I like that it returns the process object. It lets people see that there is a process object wrapping a command with a return code, all of which is good for people to be aware of.

File: base/process.jl [hunk 1]
```diff
@@ -548,68 +548,75 @@ spawn_opts_inherit(stdios::StdIOSet) = (stdios,)
...
+    _spawn(cmds, spawn_opts_swallow(args...)...; chain=chain)
 
 function eachline(cmd::AbstractCmd; chomp=nothing, keep::Bool=false)
     if chomp !== nothing
         keep = !chomp
         depwarn("The `chomp=$chomp` argument to `eachline` is deprecated in favor of `keep=$keep`.", :eachline)
     end
     stdout = Pipe()
-    processes = spawn(cmd, (DevNull,stdout,STDERR))
+    processes = _spawn(cmd, (DevNull,stdout,STDERR))
```
Comment (vtjnash) [inline base/process.jl]: Are we now unable to open a process without reaching into the internal API?

**Review by vtjnash — APPROVED** [review]

**Comment by o314** [comment]
Is there any reason to not use a positive and direct form like `run(cmd, async=true)`

rather than a double negation `run(cmd, wait=false)` ?

**Comment by StefanKarpinski** [comment]
"Don't wait" seems more obvious to me than "do run asynchronously". Similarly, "do wait" is significantly clearer than "don't run asynchronously" which is a double negative form of "run synchronously".

**Comment by o314** [comment]
I think about parallelism in terms of motion and lanes.
`async` emphasis both time and lane aspect (async ~ aside). `wait` tend to flatten both concepts.
So i prefer to use `async`. That allows me to keep in mind a graph of the whole distributed process.

I stop using mental representation in natural languages very shortly. So no need to escape the hell of understanding things like "don't run asynchronously". that's sound too much convoluted and error-prone for me.

my habits when talking about distributed processing is : use symbols that match to graph as closer as possible. Stick with them as long as possible.

**Comment by JeffBezanson** [comment]
I agree with Stefan here. While `wait = false` might kind-of, sort-of sound like a double negation, `async=false` is a direct, obvious double negation. It literally says "not-synchronous = false".

In terms of how asynchronous programming is normally described and implemented, the two primitive operations are "run asynchronously" and "wait", not "run asynchronously" and "run synchronously".

File: base/process.jl (hunk 1 above)
Comment (JeffBezanson) [reply base/process.jl]: I don't think so. Is there a case not covered by `open` and `pipeline`?

**Comment by JeffBezanson** [comment]
Hmm, this is printing an extra `hello` in the spawn test. Will investigate.

**Comment by JeffBezanson** [comment]
pkg tests seem to have network issues on OS X sometimes.

**Comment by ararslan** [comment]
This broke CoverageBase, which calls `spawn(cmd, DevNull, STDOUT, STDERR)`.

**Comment by JeffBezanson** [comment]
¯\_(ツ)_/¯

**Pull request merged**